jenkins-master/jobs/scripts/workspace/runtests.py

```
@@ -4,231 +4,123 @@
-                                content_type='link',
-                                url=http_log_url)
-        finally:
-            if job and th:
-                gecko_log_url = self.upload_s3('gecko.log')
-                if gecko_log_url:
-                    job.add_details(title='gecko.log',
-                                    value=gecko_log_url,
-                                    content_type='link',
-                                    url=gecko_log_url)
-
-                tbpl_log_url = self.upload_s3('tbpl.log')
-                if tbpl_log_url:
-                    parser = JobResultParser('tbpl.log')
-                    job.add_log_reference(os.path.basename('tbpl.log'), tbpl_log_url,
-                                          parse_status='parsed')
-                    job.add_artifact(name='text_log_summary',
-                                     artifact_type='json',
-                                     blob={'step_data': parser.failures_as_json(),
-                                           'logurl': tbpl_log_url})
-
-                job.completed(result='testfailed' if failed else 'success')
-                th.submit_results(job)
-
-    def upload_s3(self, path):
-        if not self.s3_bucket:
-            return None
+        print('Calling command to execute tests: {}'.format(command))
+        retval = subprocess.call(command)
 
+        # Save exit code into file for further processing in report submission
         try:
-            remote_filename = '%s_%s' % (str(uuid.uuid4()), os.path.basename(path))
-            return self.s3_bucket.upload(path, remote_filename)
-        except Exception as e:
-            print 'Failure uploading "%s" to S3: %s' % (path, str(e))
-            return None
+            with file('retval.txt', 'w') as f:
+                f.write(str(retval))
+        except OSError as e:
+            print('Failed to save return value: {}'.format(e))
+
+        # Delete http.log if tests were passing
```

Really? Even it is "passing", if there are silent failures or non-deterministic behaviors, it might be good to have this sitting around.